From 9fb2a8e0089c8f125c5d422eeace5c247638a42c Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Wed, 22 Apr 2020 11:55:11 -0700
Subject: [PATCH] Add adsp-fw and badpage to BUP payload

---
 bootloader/l4t_bup_gen.func | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

Index: Linux_for_Tegra/bootloader/l4t_bup_gen.func
===================================================================
--- Linux_for_Tegra.orig/bootloader/l4t_bup_gen.func
+++ Linux_for_Tegra/bootloader/l4t_bup_gen.func
@@ -195,7 +195,9 @@ _generate_bl_update_payload()
 				"${signed_dir}"/cboot_sigheader.bin.${signed_ext} cpu-bootloader 2 0 $_common_spec; \
 				"${signed_dir}"/nvtboot_sigheader.bin.${signed_ext} mb2 2 0 $_common_spec; \
 				"${signed_dir}"/spe_sigheader.bin.${signed_ext} spe-fw 2 0 $_common_spec; \
-				"${signed_dir}"/tos-trusty_sigheader.img.${signed_ext} secure-os 2 0 $_common_spec;"
+				"${signed_dir}"/tos-trusty_sigheader.img.${signed_ext} secure-os 2 0 $_common_spec; \
+				"${signed_dir}"/adsp-fw_sigheader.bin.${signed_ext} adsp-fw 2 0 $_common_spec; \
+				"${signed_dir}"/badpage_sigheader.bin.${signed_ext} badpage-fw 2 0 $_common_spec;"
 		ENTRY_LIST[kernel]=""${kernel_signed_dir}"/${kernel_image_base}_sigheader.img.${kernel_signed_ext} kernel 2 0 $_common_spec;"
 
 		if [ "${fuselevel}" == "fuselevel_nofuse" ]; then
