From 849b74836fda04a6fb24e771fd6dbd46450277eb Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Wed, 16 Jun 2021 13:14:47 -0700
Subject: [PATCH] Modify distro boot script for rauc support

Signed-off-by: Matt Madison <matt@madison.systems>
---
 include/config_distro_bootcmd.h | 52 ++++++++++++++++++++++++++++++++-
 1 file changed, 51 insertions(+), 1 deletion(-)

diff --git a/include/config_distro_bootcmd.h b/include/config_distro_bootcmd.h
index eda96ab762..a4e62abb15 100644
--- a/include/config_distro_bootcmd.h
+++ b/include/config_distro_bootcmd.h
@@ -28,7 +28,7 @@
 #define BOOTENV_SHARED_BLKDEV_BODY(devtypel) \
 		"if " #devtypel " dev ${devnum}; then " \
 			"devtype=" #devtypel "; " \
-			"run scan_dev_for_boot_part; " \
+			"run scan_dev_for_rauc_boot_part; " \
 		"fi\0"
 
 #define BOOTENV_SHARED_BLKDEV(devtypel) \
@@ -448,6 +448,7 @@
 	\
 	"boot_syslinux_conf=extlinux/extlinux.conf\0" \
 	"boot_extlinux="                                                  \
+		"env save; "						  \
 		"sysboot ${devtype} ${devnum}:${distro_bootpart} any "    \
 			"${scriptaddr} ${prefix}${boot_syslinux_conf}\0"  \
 	\
@@ -461,6 +462,7 @@
 		"fi\0"                                                    \
 	\
 	"boot_a_script="                                                  \
+		"env save; "						  \
 		"load ${devtype} ${devnum}:${distro_bootpart} "           \
 			"${scriptaddr} ${prefix}${script}; "              \
 		"source ${scriptaddr}\0"                                  \
@@ -499,6 +501,54 @@
 		"done; "                                                  \
 		"setenv devplist\0"					  \
 	\
+	"set_slot_suffix="						  \
+		"if test ${bootslot} = b; then "			  \
+			"env set bootpart_label APP_b; "	       	  \
+		"else "							  \
+			"env set bootpart_label APP; "			  \
+		"fi\0"							  \
+	\
+	"set_count="						  	  \
+		"if test ${bootslot} = a; then "			  \
+			"count=${BOOT_a_LEFT}; "			  \
+		"else "							  \
+			"count=${BOOT_b_LEFT}; "			  \
+		"fi\0"							  \
+	\
+	"decr_count="						  	  \
+		"if test ${bootslot} = a; then "			  \
+			"setexpr BOOT_a_LEFT ${BOOT_a_LEFT} - 1; "        \
+		"else "							  \
+			"setexpr BOOT_b_LEFT ${BOOT_b_LEFT} - 1; "        \
+		"fi\0"							  \
+	\
+	"scan_dev_for_rauc_boot_part=" 					  \
+		"test -n \"${BOOT_ORDER}\" || env set BOOT_ORDER \"a b\"; " \
+		"test -n \"${BOOT_a_LEFT}\" || env set BOOT_a_LEFT 3; " \
+		"test -n \"${BOOT_b_LEFT}\" || env set BOOT_b_LEFT 3; " \
+		"for bslot in ${BOOT_ORDER}; do "			  \
+			"env set bootslot $bslot; "			  \
+			"run set_slot_suffix; "				  \
+			"part number ${devtype} ${devnum} ${bootpart_label} distro_bootpart; " \
+			"if env exists distro_bootpart; then "		  \
+				"run set_count; if test $count -gt 0; then "	\
+					"echo \"Found valid slot ${bootslot}, ${count} attempts remaining\"; " \
+					"run decr_count; "			\
+					"if fstype ${devtype} "                 \
+						"${devnum}:${distro_bootpart} " \
+						"bootfstype; then "             \
+						"run scan_dev_for_boot; "       \
+					"fi; "                                  \
+				"fi; "                                    \
+				"setenv distro_bootpart; "		  \
+			"fi; "						  \
+			"env set bootslot; "				  \
+		"done; "						  \
+		"echo \"No valid slot found, resetting counts\"; "	  \
+		"env set BOOT_a_LEFT 3; "				  \
+		"env set BOOT_b_LEFT 3; "				  \
+		"env save; reset\0"					  \
+	\
 	BOOT_TARGET_DEVICES(BOOTENV_DEV)                                  \
 	\
 	"distro_bootcmd=" BOOTENV_SET_SCSI_NEED_INIT                      \
