From 99788f8abf96b1464162390da11632fcd8d58c93 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Tue, 29 Sep 2020 10:05:08 -0700
Subject: [PATCH] Align group names with systemd requirements

Signed-off-by: Matt Madison <matt@madison.systems>

---
 group.master | 1 +
 1 file changed, 1 insertion(+)

diff --git a/group.master b/group.master
index 5b62284..d522c90 100644
--- a/group.master
+++ b/group.master
@@ -39,4 +39,5 @@ staff:*:50:
 games:*:60:
 shutdown:*:70:
 users:*:100:
+render:*:134:
 nogroup:*:65534:
